src/app/CommandResponseSender.cpp

```diff
@@ -157,12 +171,52 @@ void CommandResponseSender::MoveToState(const State aTargetState)
 void CommandResponseSender::Close()
 {
     MoveToState(State::AllInvokeResponsesSent);
-    mCloseCalled = true;
-    if (mResponseSenderDoneCallback)
+    mpCallback->OnDone(*this);
+}
+
+void CommandResponseSender::OnInvokeCommandRequest(Messaging::ExchangeContext * ec, System::PacketBufferHandle && payload,
+                                                   bool isTimedInvoke)
+{
+    VerifyOrDieWithMsg(ec != nullptr, DataManagement, "Incoming exchange context should not be null");
+    VerifyOrDieWithMsg(mState == State::ReadyForInvokeResponses, DataManagement, "state should be ReadyForInvokeResponses");
+
+    // NOTE: we already know this is an InvokeCommand Request message because we explicitly registered with the
+    // Exchange Manager for unsolicited InvokeCommand Requests.
+    mExchangeCtx.Grab(ec);
+    mExchangeCtx->WillSendMessage();
+
+    // Grabbing Handle to prevent OnDone from being called before OnInvokeCommandRequest returns.
```

Comment:
I am not seeing a CommandResponseSender::OnDone.  Am I just missing it?

Reply:
It is in the header. I moved out the implementation into the .cpp file now so it should be easier to find